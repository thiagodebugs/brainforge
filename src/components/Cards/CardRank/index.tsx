interface CardRankProps {
  cardRank: {
    title: string;
    tag: string;
    process: {
      id: number;
      title: string;
      progress: string;
    }[];
  };
}

import style from "./style.module.css";

export default function CardRank({ cardRank }: CardRankProps) {
  return (
    <>
      <div className={style.cardRank}>
        <div className={style.cardRankTitle}>
          <h1>{cardRank.title}</h1>
          <h3>{cardRank.tag}</h3>
        </div>
        {cardRank.process.map((process) => (
          <div key={process.id} className={style.cardRankProcess}>
            <div className={style.cardRankProcessTitle}>
              <h2>{process.title}</h2>
              <p>{process.progress}</p>
            </div>
            <div className={style.progressBar}>
              <div
                style={
                  process.progress.includes("%")
                    ? { width: process.progress }
                    : {
                        width: `calc(20% * ${process.progress.replace(
                          "/5",
                          ""
                        )})`,
                      }
                }
              ></div>
            </div>
          </div>
        ))}
      </div>
    </>
  );
}
