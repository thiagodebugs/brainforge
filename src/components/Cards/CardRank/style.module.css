.cardRank {
  @apply flex flex-col bg-neutral-800 rounded-xl pt-4 pb-8 px-6 space-y-4;
}

.cardRankTitle {
  @apply flex justify-between items-center pb-2 mb-2 border-b border-zinc-600 font-semibold text-lg;
}

.cardRankTitle h3 {
  @apply text-[#26c666] text-2xl;
}
.cardRankProcess {
  @apply flex flex-col space-y-4;
}

.cardRankProcessTitle {
  @apply flex justify-between items-center;
}

.cardRankProcessTitle h2 {
  @apply text-sm font-semibold;
}

.cardRankProcessTitle p {
  @apply text-xs;
}

.progressBar {
  @apply w-full h-1 bg-neutral-700 rounded-full;
}

.progressBar div {
  @apply block h-1 bg-[#26c666];
}
