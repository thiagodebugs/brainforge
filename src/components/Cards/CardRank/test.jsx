import { render } from "@testing-library/react";
import CardRank from "./";

const props = {
  title: "Test",
  tag: "V3",
  process: [
    {
      id: 1,
      title: "Test",
      progress: "2/5",
    },
  ],
};

describe("CardRank", () => {
  it("should render the card rank", () => {
    const { container } = render(<CardRank cardRank={props} />);

    expect(container).toMatchSnapshot();
  });
});
