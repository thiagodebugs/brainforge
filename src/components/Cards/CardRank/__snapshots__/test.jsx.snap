// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardRank should render the card rank 1`] = `
<div>
  <div
    class="cardRank"
  >
    <div
      class="cardRankTitle"
    >
      <h1>
        Test
      </h1>
      <h3>
        V3
      </h3>
    </div>
    <div
      class="cardRankProcess"
    >
      <div
        class="cardRankProcessTitle"
      >
        <h2>
          Test
        </h2>
        <p>
          2/5
        </p>
      </div>
      <div
        class="progressBar"
      >
        <div
          style="width: calc(20% * 2);"
        />
      </div>
    </div>
  </div>
</div>
`;
